### ISSUE TITLE
xi18n should parse index.html and build should fill in the translated texts in index.html

### ISSUE THREAD

**Issue by areijngoudt** [angular/angular-cli#8947]
### Versions
```
Angular CLI: 1.6.0
Node: 8.9.1
OS: darwin x64
Angular: 5.1.0
... animations, common, compiler, compiler-cli, core, forms
... http, platform-browser, platform-browser-dynamic, router
... service-worker

@angular/cdk: 5.0.0
@angular/cli: 1.6.0
@angular/material: 5.0.0
@angular-devkit/build-optimizer: 0.0.35
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.41
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.0
@schematics/angular: 0.1.10
@schematics/schematics: 0.0.10
typescript: 2.4.2
webpack: 3.10.0
```

### Repro steps

* in index.html, use the i18n marker like <p i18n>please wait...</p>
* run ng xi18n
* the 'please wait' does not popup in the gererated .xlf files
* run ng build ng build -prod --aot --output-path=dist/de --base-href /de/ --i18n-file=src/i18n/messages.de.xlf --i18n-format=xlf --locale=de; 

### Observed behavior
```
<!-- Normally this includes a stack trace and some more information. -->
```

**Comment by filipesilva** [comment]
`ng xi18n` operates over components of your Angular application, but `index.html` is not part of it actually. It's not a component template.

I don't think this is functionality we can provide but @ocombe might know more.

**Comment by areijngoudt** [comment]
wel, the entry and name index.html are in the .angular-cli.json. Also, the cli can inject the base href using the --base-href switch

**Comment by ocombe** [comment]
it will be possible as part of universal (it's on the roadmap)

**Comment by rmedaer** [comment]
If I can't use server rendering, is there a plan to inject translated string in `index.html` or other static file ?

**Comment by lpsBetty** [comment]
if parsing the `index.html` would be possible, we could also set the right language attribute in the HTML-tag.

**Comment by luchillo17** [comment]
What's the status on this? I'm trying to set up locale-specific `index.html` to change some tags specific to the locale, like the `title`, the HTML `lang` attribute, canonical link, and others, but can't rely on the i18n translation.

Was trying to make multiple `index.html` files, 1 per locale, but when I run with `--localize` flag it ignores the `angular.json` config about which `index.html` to use.

Looks like this, building with `--configuration=es-CO` works, but using `--localize` to build all locales at once doesn't work, I think I might have to file a bug:
![image](https://user-images.githubusercontent.com/9698639/81490602-fe9a1480-9249-11ea-8f57-1696ea8c161e.png)

**Comment by luchillo17** [comment]
#17684 was closed because including the `index.html` in the localization process would be a better solution than providing a way to compile multiple `configurations` at once, or providing the i18n a way to configure the `index.html` to use for each locale.

For now, working around it by creating a `configuration` for each locale and building all of them through gulp script in series.

**Comment by Amthieu** [comment]
Has there been any updates regarding this issue? It would be a great improvement being able to translate the index.html

**Comment by Dikman** [comment]
Yes, I also have such wish in a project on Angular 11. Is it exists a solving way?

**Comment by timothyBrake** [comment]
We need this feature to specify i18n og-meta tags in index.html.
Use case would be a customer application where customers must always log in BUT can receive e.g. an invoice link using WhatsApp that goes to the page that displays the invoice for the user. Company would send a link using WhatsApp and user would

1. click on the link in WhatsApp
2. log in (if not already)
3. Web application opens the page and users can see the invoice

A very basic use case in 2022 and we of course need this to be language specific to have a nice description, title and icon in WhatsApp

Angular should have a feature for this use case in 2022:
`<meta property="og:description" i18n-content="@@og.description" content="Welcome to the customer application of AweSomeCompany. " />`

As others have stated: seems like the Angular CLI could easily process the index.html and put the correct value in there as its a valid uses case using i18n.
Is there an Angular way, an official Angular workaround way of doing this or is this use case not going to be recognised?

Adding @AndrewKushnir as he seems to be in charge of i18n at Angular.

Thanks.

**Comment by BartCorremans** [comment]
Would also like to see this so we can translate our `noscript` message.

**Comment by xanderknol** [comment]
Can this feature request be re-evaluated?

For us this would be really useful so we can set different script url's for different languages.
Currently there is no proper way of doing this.

**Comment by zajm** [comment]
I also just stumbled over this and would greatly appreciate a fix.

**Comment by csisy** [comment]
I think the i18n configuration of the `angular.json` should be extended. Consider the following use-case:

We're building a website which is targeted for a specific country. It has some features that only make sense for people who can speak the language of that country. So we'd like to drop some functionality (at compile-time!) from the other localized versions of the app. This would be possible with an `environment`-like setup (with file replacements) and use a compile-time constant to conditionally import that module.

However, as it was discussed before, the current localization process is not designed in a way that it can re-build the app for different locales. (Technically this issue is similar to https://github.com/angular/angular-cli/issues/17684)

We've tried the configuration-way: instead of providing the `--localize` flag in the CLI, we created multiple configurations that set the appropriate `locale`. However, when the build starts, the output directory is cleared which makes this approach uncomfortable (the built version for locale A must be copied somewhere else before the build is started for locale B).

Is there any plan to extend the i18n system?
